### ISSUE TITLE
better data source for election data?

### ISSUE THREAD

**Issue by happyjack27** [happyjack27/autoredistrict#3]
one thing i've struggled with a lot is getting good clean data on vote counts at the voting tabulation district level.  I've been able to get clean vtd-level shapefiles and population data from census.gov, but election data seems to be all over the place and varied not only in format, but quality.  e.g. i've found some shapefiles with so many geometric areas that they are unusable, etc.

would like nice clean, centralized way to import this data, despite the apparent great variety in source, format, and quality of data depending on state.

**Comment by carlschroedl** [comment]
Oh brother, it's disappointing that the election data isn't available more consistently. At work we have a few apps that aggregate data from multiple independent organizations. It can be a lot of work to iron out the nuances of the different data providers. It always helps to have a machine-verifiable representation of what you would like in the end. So, for example, if you were using various XSLTs (one per independent data provider) to transform XML web service responses into a common format, it would be important to define an XSD for the common format, and perhaps some code that could perform higher-level validation logic on the transformed output. In this case, a good start would be pursuing the definition of a machine-verifiably "good" target.

**Comment by carlschroedl** [comment]
I couldn't find anything relevant on a quick look at data.gov, but I suspect such a consolidated dataset would live in there if it exists. You might be able to supply better search terms.

**Comment by happyjack27** [comment]
yeah, a consistent formal would be nice.  not a big fan of xml though, it's
kludgy, bulky, inelegant... json is much better.
and would rather just have it in a shapefile, or failing that,
tab-delimited, keyed on geoid at the vtd-level, matching some standard
datasource for the shapefile (such as census.gov)

i have two data source in the "file" menu, havard election data archive and
another one, they are kind of meant to be a central store, but
unfortunately they're rather hit-or-miss on a number of levels.  a bit
disappointed.  was expecting better quality from, you know, harvard.

**Comment by carlschroedl** [comment]
Yeah, sorry if I was unclear -- I'm not pushing for any particular serialization of the data, I was just drawing on an example that happens to use XML. I'm suggesting the pursuit of some sort of analog in the new context of this voting data.

**Comment by happyjack27** [comment]
data sources for all but election data:

http://www.census.gov/rdo/pdf/StrengthInNumbers2010.pdf

https://www.census.gov/rdo/data/
